File: 1-two-sum/two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> prev = new HashMap<Integer, Integer>();
        int[] result = new int[2];
        for(int i = 0; i < nums.length; i++){
            if(prev.get(target - nums[i]) != null){
                result[0] = prev.get(target - nums[i]);
                result[1] = i;
                return result;
            }
            else{
                prev.put(nums[i],i);
            }
        }
        return result; 
    }
}

File: 1-two-sum/two-sum.cpp
#include<map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> prev;
        vector<int> temp;
        for(int i = 0; i < nums.size(); i++){
            if(prev.count(target - nums[i])){
                
                temp.push_back(prev[target - nums[i]]);
                temp.push_back(i);
                return temp;
            }
            else{
                prev[nums[i]] = i;
            }
        }
        return temp;
    }
};